lib/fs.js

```diff
@@ -837,7 +843,13 @@ function ftruncate(fd, len = 0, callback) {
 function ftruncateSync(fd, len = 0) {
   validateInt32(fd, 'fd', 0);
   validateInteger(len, 'len');
-  len = MathMax(0, len);
+
+  if (len < 0) {
+    throw new ERR_FS_FILE_TOO_SMALL(len);
```

Comment:
I would use `ERR_INVALID_ARG_VALUE`  here – it’s not the file that has a negative length, it is an invalid argument to the function here.

Likewise, using `kIoMaxLength`/`ERR_FS_FILE_TOO_LARGE` seems a bit odd here. The 2 GB limit is because that is the maximum buffer size supported, but Node.js generally does support way larger files than that (just not reading or writing to them in multi-GB chunks). We should not fail just because `len` exceeds this smaller limit, and only fail if it exceeds the actual limit after which the operation would fail or do the wrong thing.

Reply:
Makes sense!

So, that should solve
```
  if (len < 0) {
    throw new ERR_INVALID_ARG_VALUE(len);
  }
```

I'll update the PR